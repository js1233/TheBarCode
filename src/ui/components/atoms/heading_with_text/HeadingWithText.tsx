import React from "react";
import { StyleProp, View, ViewStyle } from "react-native";
import {
  AppLabel,
  TEXT_TYPE
} from "ui/components/atoms/app_label/AppLabel";
import { FONT_SIZE, STRINGS } from "config";

interface OwnProps {
  containerStyle?: StyleProp<ViewStyle>;
  headingText: string;
  text?: string;
  headingStyle?: StyleProp<ViewStyle>;
  headingTextType?: TEXT_TYPE;
  textStyle?: StyleProp<ViewStyle>;
}

type Props = OwnProps;

export const HeadingWithText = React.memo<Props>(
  ({
    containerStyle,
    text,
    headingText,
    headingStyle,
    textStyle,
    headingTextType = TEXT_TYPE.SEMI_BOLD
  }) => {
    return (
      <View testID={"HEADING_WITH_TEXT"} style={containerStyle}>
        <AppLabel
          text={headingText}
          textType={headingTextType}
          style={[{ fontSize: FONT_SIZE.lg }, headingStyle]}
        />
        <AppLabel
          text={text ?? STRINGS.common.label_n_a}
          numberOfLines={0}
          style={[{ fontSize: FONT_SIZE.sm }, textStyle]}
        />
      </View>
    );
  }
);
