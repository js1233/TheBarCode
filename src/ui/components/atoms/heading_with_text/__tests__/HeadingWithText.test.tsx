import { render } from "@testing-library/react-native";
import React from "react";
import { HeadingWithText } from "ui/components/atoms/heading_with_text/HeadingWithText";

test("check render properly", () => {
  const { getByTestId } = render(
    <HeadingWithText
      headingText={"Lorem Ipsum"}
      text={
        "Lipsum as it is sometimes known, is dummy text used in laying out print."
      }
    />
  );
  const headingWithText = getByTestId("HEADING_WITH_TEXT");
  expect(headingWithText).toBeDefined();
});

test("Snapshot testing", () => {
  const rendered = render(
    <HeadingWithText
      headingText={"Lorem Ipsum"}
      text={
        "Lipsum as it is sometimes known, is dummy text used in laying out print."
      }
    />
  ).toJSON();
  expect(rendered).toMatchSnapshot();
});
