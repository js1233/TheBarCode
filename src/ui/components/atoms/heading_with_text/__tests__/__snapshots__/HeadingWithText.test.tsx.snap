// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing 1`] = `
<View
  testID="HEADING_WITH_TEXT"
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "fontFamily": "ProximaNova-Semibold",
        },
        Object {
          "color": "#171717",
          "fontSize": 27,
        },
        Object {
          "lineHeight": 45,
        },
        Array [
          Object {
            "fontSize": 32,
          },
          undefined,
        ],
      ]
    }
  >
    Lorem Ipsum
  </Text>
  <Text
    numberOfLines={0}
    style={
      Array [
        Object {
          "fontFamily": "ProximaNova-Regular",
        },
        Object {
          "color": "#171717",
          "fontSize": 27,
        },
        Object {
          "lineHeight": 38,
        },
        Array [
          Object {
            "fontSize": 27,
          },
          undefined,
        ],
      ]
    }
  >
    Lipsum as it is sometimes known, is dummy text used in laying out print.
  </Text>
</View>
`;
